lib/SILOptimizer/Mandatory/SemanticARCOpts.cpp

```diff
@@ -79,75 +79,67 @@ class LiveRange {
 
 LiveRange::LiveRange(SILValue value)
     : destroys(), generalForwardingInsts(), unknownConsumingUsers() {
+  // We know that our silvalue produces an @owned value. Look through all of our
+  // uses and classify them as either consuming or not.
   SmallVector<Operand *, 32> worklist(value->getUses());
-
   while (!worklist.empty()) {
     auto *op = worklist.pop_back_val();
 
-    // Skip type dependent operands.
-    if (op->isTypeDependent())
+    // Skip type dependent operands and operands whose incoming value is not
+    // owned.
+    if (op->isTypeDependent() ||
+        op->get().getOwnershipKind() != ValueOwnershipKind::Owned)
```

Comment:
This is kind of mysterious. Isn't `value` known to be `owned`. If so, how does a `guaranteed` operand ever end up on the worklist?

Reply:
I think this was intended to actually be an assert in case someone updated the code and added such a thing. I am going to convert this line to an assert.